distributed/scheduler.py

```diff
@@ -5461,7 +5480,7 @@ def add_plugin(self, plugin=None, idempotent=False, name=None, **kwargs):
 
         """
         if isinstance(plugin, type):
-            plugin = plugin(self, **kwargs)
+            plugin = plugin(self, **kwargs)  # type: ignore
```

Comment:
This could probably be fixed by capturing scheduler plugin kwargs in a `ParamSpec`  and making a `TypeVar` for the `add_plugin` signature. I don't think we should actually do that here, though, since `ParamSpec` is in `typing_extensions~=3.10`, which is not universally supported yet.

Reply:
mypy is right complaining here. This is genuinely wrong; most plugins DO NOT accept a scheduler positional arg.
See #5394 for proper fix